### ISSUE TITLE
2.0 models are not producing a good output when using animator.

### ISSUE THREAD

**Issue by LIQUIDMIND111** [Animator-Anon/Animator#42]
HELLO GOOD SIR!! Just wanted to let you know that the animator script works VERY WELL with 1.5 models, but apparently with 2.0 models, after 5 seconds of animation, USING THE SAME PARAMETERS I  HAVE USED BEFORE, it starts to literally break apart into particles, lol.... i dont mind using 1.5 models, it does not matter if its a 768x768 models, but if its 2.0 or 2.1 the animations will break like this....
![20221212090021_00000](https://user-images.githubusercontent.com/112510084/207069776-4c3dce10-b1c8-467b-b42d-9984c727ac8e.png)

TO THIS
![20221212090021_00399](https://user-images.githubusercontent.com/112510084/207069822-01873159-9aec-41a0-847d-e8ae3b8392c6.png)

i dont mind if it cant be fixed..... just wanted to let you know ion case some people come in the future to complain, lol

**Comment by Animator-Anon** [comment]
I haven't even tried these models yet, i'm a bit behind the times. I'll look into it, but I suspect it isn't anything I am doing. Thanks for heads up.

**Comment by LIQUIDMIND111** [comment]
no worries man!!! have a great Christmas. your v6 is good enough, we appreciate it a lot! do you have a Facebook or twitter page?

**Comment by Animator-Anon** [comment]
I do not, this is the only place I exist!

**Comment by LIQUIDMIND111** [comment]
> I do not, this is the only place I exist!

Buddy, look at this animation i did in deforum using the new 2.1 models with a custom model 
https://youtu.be/yTWNrHFMNnU


I WANT TO DO THE SAME WITH YOUR SCRIPT as i was doing before, but now, when i try again with animator v6 on 2.0 models, i get this result....

https://user-images.githubusercontent.com/112510084/208271468-7ed8a957-358f-4d1c-b2fe-96bb06c3d13c.mp4


 is very rare, compared to animator 6 using 1.5 models. Something is going on with the script when it renders models made with 2.0 2.1, and I LOVE YOUR SCRIPT, please let me know if you can't take a look at the code and if something needs to be upgraded for the 2.0 models!!!! Your script is SO EASY TO ACCESS, as good as deforum.... awesome! let me know any updates eventually, my brother!!

**Comment by LIQUIDMIND111** [comment]
JUST A NOTE, AI use is going to be HUGE and your script is the ONLY scrip besides deforum, that anyone can do animation so GOOD, please try to update the code to accept this new models!!! i will try to send some donations or tips if you send me your BUY ME A COFFEE PAGE or whatever you use!!!! i will spread it to the community for animators and influence them in donating also!!! this models are amazing!!! Photorealistic!! look at that!!! we need you buddy!!!!!

**Comment by Animator-Anon** [comment]
Alright I'm download a 2.0 model now.
In the mean time check out the extension version of this script. Probably broken in many ways, but should more or less work the same. Just a bit easier because you don't have to supply an initial image in img2img mode now, It can generate one itself.

**Comment by Animator-Anon** [comment]
[video.webm](https://user-images.githubusercontent.com/114563845/208284224-10e65b6a-0b07-495f-b9d1-eff4f1e5c0ac.webm)
Here is one I just made, stock 512x512 SD 2.0 with the basic inference .yaml file.

**Comment by LIQUIDMIND111** [comment]
> [video.webm](https://user-images.githubusercontent.com/114563845/208284224-10e65b6a-0b07-495f-b9d1-eff4f1e5c0ac.webm) Here is one I just made, stock 512x512 SD 2.0 with the basic inference .yaml file. Also used my extension, which is all ill probably update going forward: https://github.com/Animator-Anon/animator_extension

bro if you did that animation with 2.0 models, then its fixed..... so i need to upgrade to EXTENSION TAB?

**Comment by LIQUIDMIND111** [comment]
> Alright I'm download a 2.0 model now. In the mean time check out the extension version of this script. Probably broken in many ways, but should more or less work the same. Just a bit easier because you don't have to supply an initial image in img2img mode now, It can generate one itself.

oh ok very nice then, but when you mean BROKEN, you mean that it can show errors during an animation? or what can happens?

**Comment by LIQUIDMIND111** [comment]
> > Alright I'm download a 2.0 model now. In the mean time check out the extension version of this script. Probably broken in many ways, but should more or less work the same. Just a bit easier because you don't have to supply an initial image in img2img mode now, It can generate one itself.
> 
> oh ok very nice then, but when you mean BROKEN, you mean that it can show errors during an animation? or what can happens?

if i add the extnesion can i also use the older version sin any case it has isues?

**Comment by Animator-Anon** [comment]
Yeah they can co-exist fine.
I'll add instructions to install, you might be able to do it from the UI, I haven't tested that yet. Either way its a different repository, and a different set of files that go into the extensions folder.
By broken the code has been re-arranged and I might not have tested everything.
Some features might work slightly differently when it comes to the edge cases.
BY fixed, I didn't do anything to fix your specific issue. I wonder if its a problem with your model. Try a fresh 512x512 SD2.0 download from hugging face with the [v2-inference.yaml](https://github.com/Stability-AI/stablediffusion/blob/main/configs/stable-diffusion/v2-inference.yaml) from github.

**Comment by LIQUIDMIND111** [comment]
> Yeah they can co-exist fine. I'll add instructions to install, you might be able to do it from the UI, I haven't tested that yet. Either way its a different repository, and a different set of files that go into the extensions folder. By broken the code has been re-arranged and I might not have tested everything. Some features might work slightly differently when it comes to the edge cases. BY fixed, I didn't do anything to fix your specific issue. I wonder if its a problem with your model. Try a fresh 512x512 SD2.0 download from hugging face with the [v2-inference.yaml](https://github.com/Stability-AI/stablediffusion/blob/main/configs/stable-diffusion/v2-inference.yaml) from github.

ok i did a GIT CLONE of the new exntension

C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\extensions>git clone https://github.com/Animator-Anon/animator_extension
Cloning into 'animator_extension'...
remote: Enumerating objects: 21, done.
remote: Counting objects: 100% (21/21), done.
remote: Compressing objects: 100% (20/20), done.
Receiving objects:  61% (13/21)sed 0 (delta 0), pack-reused 0
Receiving objects: 100% (21/21), 21.34 KiB | 1.78 MiB/s, done.
Resolving deltas: 100% (3/3), done.

C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\extensions>

WILL TRY SOON!!!

**Comment by LIQUIDMIND111** [comment]
> > Yeah they can co-exist fine. I'll add instructions to install, you might be able to do it from the UI, I haven't tested that yet. Either way its a different repository, and a different set of files that go into the extensions folder. By broken the code has been re-arranged and I might not have tested everything. Some features might work slightly differently when it comes to the edge cases. BY fixed, I didn't do anything to fix your specific issue. I wonder if its a problem with your model. Try a fresh 512x512 SD2.0 download from hugging face with the [v2-inference.yaml](https://github.com/Stability-AI/stablediffusion/blob/main/configs/stable-diffusion/v2-inference.yaml) from github.
> 
> ok i did a GIT CLONE of the new exntension
> 
> C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\extensions>git clone https://github.com/Animator-Anon/animator_extension Cloning into 'animator_extension'... remote: Enumerating objects: 21, done. remote: Counting objects: 100% (21/21), done. remote: Compressing objects: 100% (20/20), done. Receiving objects: 61% (13/21)sed 0 (delta 0), pack-reused 0 Receiving objects: 100% (21/21), 21.34 KiB | 1.78 MiB/s, done. Resolving deltas: 100% (3/3), done.
> 
> C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\extensions>
> 
> WILL TRY SOON!!!

let me know if i need to do something else

**Comment by LIQUIDMIND111** [comment]
i had a loading error, look

Error loading script: animator.py
Traceback (most recent call last):
  File "C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\modules\scripts.py", line 195, in load_scripts
    module = script_loading.load_module(scriptfile.path)
  File "C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\modules\script_loading.py", line 13, in load_module
    exec(compiled, module.__dict__)
  File "C:\TEMP\Stable diff\Autom111\stable-diffusion-webui\extensions\animator_extension\scripts\animator.py", line 14, in <module>
    from extensions.animator.scripts.functions import prepwork, sequential, loopback, export
ModuleNotFoundError: No module named 'extensions.animator'

ANY IDEAS?

**Comment by LIQUIDMIND111** [comment]
ok for the moment is LOADING because i renamed the folder to ANIMATOR only

**Comment by LIQUIDMIND111** [comment]
ok tab is loaded..... i will test later when i get back from walking he dogs... will let you know if it breaks or not.......

**Comment by LIQUIDMIND111** [comment]
ok i cant find the input for zoom, x y rotation, etc......

**Comment by LIQUIDMIND111** [comment]
ok im lost..... i cant find where to write the prompt, or how to import an image, you said it can create one but i dont see a prompt input... and i just click generate and had a back pack, please. let me know the NEW steps now...

**Comment by LIQUIDMIND111** [comment]
and whats the output folder now?

**Comment by LIQUIDMIND111** [comment]
https://user-images.githubusercontent.com/112510084/208327054-693415c6-e9b0-446a-947b-17bb2c989997.mp4

buddy, it wont come out as before, i have tried to change everything and nothing changes this output, whats happening?

**Comment by LIQUIDMIND111** [comment]
OK MATE, here is my feedback -  its not doing a seamless animation even with seed marching or seed travel turn off.

2) animations, after the 1st frame, basically uses fewer steps to do the next frames, and HERE it will always do the same step as the beginning. not sure of that could be a problem 

this setting should be OFF depending 

With img2img, do exactly the amount of steps the slider specifies (normally you'd do less with less denoising).

**Comment by Animator-Anon** [comment]
Did you turn loopback off? If you turn this off, it will generate a new frame every time rather than re-processing the previous frame. Without prompts or screenshots I can see.

**Comment by LIQUIDMIND111** [comment]
> Did you turn loopback off? If you turn this off, it will generate a new frame every time rather than re-processing the previous frame. Without prompts or screenshots I can see.
> 
> If you do have issues with the picture jumping around using seed travel, I found found that doesn't work well with ascension diffusers, or what ever they are called. The ones ending in _a. Euler by itself should work fine. That might be unrelated to what you have done here.

ahh yes i remove the LOOPBACK because i thought it was something else.... hum, let me try again... one more thing, where are the rotation, X Y and ZOOM input??

**Comment by LIQUIDMIND111** [comment]
![Screenshot 2022-12-18 231446](https://user-images.githubusercontent.com/112510084/208340472-913f7e62-4ae8-4ade-8436-31588b09fdbb.jpg)


also its not detecting the seed i type.... and the steps need to have an option to choose less than the original

**Comment by Animator-Anon** [comment]
You'll have to use the transform keyframe at time 0.
Currently no way to directly load an initial image, other than using a prop. So set a prop folder in the settings, drag it in there and call it up at time 0. I guess the idea was an initial image would be generated from the prompts.
Normally I would generate a picture in the txt2img tab, and then drag that into the img2img tab to start working on it. Just removed a step, but i guess it also removes some flexibility.

**Comment by LIQUIDMIND111** [comment]
Outputs are broken mate.... so sad, i believe something in the code is NOT compatible with 2.0 custom models, since you actually made a working animation that KEPT QUALITY and coherence all the way,,, look at these videos,,,, they get broken apart.....

https://user-images.githubusercontent.com/112510084/208343633-749b6669-b8e2-40b0-9313-ac04c6e0fbb4.mp4

https://user-images.githubusercontent.com/112510084/208343654-553ecfd8-b3f7-4d14-88d3-25e775ae09c7.mp4


 anyway, yes the image input would be awesome.... not sure why this happens but i hope it can be fixed in the future, since it really works perfect with CUSTOM MODELS made with 1.5 - but if you tested it with a 2.0 base model, and it worked, maybe it means that custom models on 2.0 has issues...... 

what do you think?

**Comment by LIQUIDMIND111** [comment]
THIS IS A SAMPLE of a custom model made with 1.5 on animator 6, look how perfect it shows....
https://user-images.githubusercontent.com/112510084/208344202-e90abc2f-fabc-4343-a391-6d288f6b797f.mp4

**Comment by Animator-Anon** [comment]
Working on image input.
Now that just looks like the seed is not changing., and it is img2img'ing the same picture.
Check the .csv in the output folder to check if the seed is changing, or if you have seed travel turned on, that there is a start and subseed and the seed strength is increasing.

**Comment by LIQUIDMIND111** [comment]
> Working on image input. Now that just looks like the seed is not changing., and it is img2img'ing the same picture. Check the .csv in the output folder to check if the seed is changing, or if you have seed travel turned on, that there is a start and subseed and the seed strength is increasing.

ok seed travel i turn off, let me check the csv

**Comment by LIQUIDMIND111** [comment]
> 

well, seems it IS changing seeds, not sure, here is part of the CSV

*pos1,neg1,pos2,neg2,prompt,denoise,noise,x_shift,y_shift,zoom,rotation,cfg_scale,seed_start,seed_end,seed_str,pos_prompt,neg_prompt
0,,,,,1.0,0.29,0.1,0.06666666666666667,0.06666666666666667,1.0,0.06666666666666667,11.0,796368297.0,,0,"Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:1.0 AND Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:0.0","blur,:1.0 AND blur,:0.0"
1,,,,,0.9955555555555555,0.29,0.1,0.07609502161608059,0.0407516156348888,0.9997956344236903,0.0630440789766903,11.0,796368298.0,,0,"Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:0.9955555555555555 AND Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:0.004444444444444473","blur,:0.9955555555555555 AND blur,:0.004444444444444473"
2,,,,,0.9911111111111112,0.29,0.1,0.08637427625634607,0.015610481064151359,0.999593488067007,0.059586154363531034,11.0,796368299.0,,0,"Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:0.9911111111111112 AND Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:0.008888888888888835","blur,:0.9911111111111112 AND blur,:0.008888888888888835"
3,,,,,0.9866666666666667,0.29,0.1,0.0975044305874631,-0.008756737045545715,0.9993935609299498,0.05629289282718886,11.0,796368300.0,,0,"Photo of a girl in the forest, 8k, Canon EOS Mark 70D, glossy surfaces, subsurface scattering, XxXUltxraDrexamXxX,:0.9866666666666667 AND Photo of a girl **

**Comment by LIQUIDMIND111** [comment]
ok how do i check this also?

 that there is a start and subseed and the seed strength is increasing.

**Comment by LIQUIDMIND111** [comment]
i always use euler a

**Comment by Animator-Anon** [comment]
So your seed was stuck at 11. The end seed was changing for some reason, but not used because seed travel was disabled.
See if you can update via the extensions tab and try stuff again. It might have been a bug but I am not sure how that happened.

**Comment by LIQUIDMIND111** [comment]
> So your seed was stuck at 11. The end seed was changing for some reason, but not used because seed travel was disabled. See if you can update via the extensions tab and try stuff again. It might have been a bug but I am not sure how that happened.

the 11 is CFG i believe, or what are you referring? ok i will update later buddy thanks a lot man, you are a great person, :D

**Comment by Animator-Anon** [comment]
Ah you are right, header for time was missing.

**Comment by LIQUIDMIND111** [comment]
hello mate!!! Merry xmas! i found out what's happening with animations.... it has to do with the pixels amount = more than 500000 pixels will give errors on MIDAS adabin weights - it happens in deforum also, SOMETIMES i get to manage and DON'T get errors, in THEIR CASE you get BLACK FRAMES, in your case you get RARE NOISE, but due to the same errors in MIDAS.... so, the BEST thing to do is DONT GO HIGHER than 768x640 on  your script and BOOM, works well!!! :D

**Comment by Animator-Anon** [comment]
Good you figured it out! Not sure I am doing anything specific to chew up vram, I assume that is what is causing those issues.
